File: benchmarks/eval_headtohead/corpus/go/reporting.go
// Reporting — a LEXICAL DISTRACTOR module (§3 pre-registration).
// Mentions ProcessOrder in a comment and a string, and defines
// substring-colliding names (PreprocessOrder, ProcessOrders). None of these
// are real call sites of ProcessOrder.
package order

// NOTE: ProcessOrder is the hot path; report on its throughput here.
const Label = "ProcessOrder latency"

func PreprocessOrder(order map[string]string) map[string]string {
	out := map[string]string{}
	for k, v := range order {
		out[k] = v
	}
	return out
}

func ProcessOrders(orders []map[string]string) []map[string]string {
	out := make([]map[string]string, 0, len(orders))
	for _, o := range orders {
		out = append(out, PreprocessOrder(o))
	}
	return out
}

File: benchmarks/eval_headtohead/corpus/rust/reporting.rs
//! Reporting — a LEXICAL DISTRACTOR module (§3 pre-registration).
//! Mentions process_order in a comment and a string, and defines
//! substring-colliding names (preprocess_order, process_orders). None of these
//! are real call sites of core::process_order.

use std::collections::HashMap;

// NOTE: process_order is the hot path; report on its throughput here.
pub const LABEL: &str = "process_order latency";

pub fn preprocess_order(order: &HashMap<String, String>) -> HashMap<String, String> {
    order.clone()
}

pub fn process_orders(orders: Vec<HashMap<String, String>>) -> Vec<HashMap<String, String>> {
    orders.iter().map(preprocess_order).collect()
}
